File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(n):
    return pow(sum(range(1, n+1)), 2)


def sum_of_squares(n):
    return sum(map(lambda x: pow(x, 2), range(1, n+1)))


def difference_of_squares(n):
    return square_of_sum(n) - sum_of_squares(n)

File: solutions/csharp/difference-of-squares/1/DifferenceOfSquares.cs
public static class DifferenceOfSquares
{
    public static int CalculateSquareOfSum(int max)
    {
        ArgumentOutOfRangeException.ThrowIfNegative(max);
        var sum = max * (max + 1) / 2;
        return sum * sum;
    }

    public static int CalculateSumOfSquares(int max)
    {
        ArgumentOutOfRangeException.ThrowIfNegative(max);
        return max * (max + 1) * (2 * max + 1) / 6;
    }

    public static int CalculateDifferenceOfSquares(int max) =>
        CalculateSquareOfSum(max) - CalculateSumOfSquares(max);
}